couchdb/server/__init__.py

```diff
@@ -234,6 +234,25 @@ def respond(self, data):
         """
         return self._respond(data)
 
+    def log(self, message):
+        """Log message to CouchDB output stream.
+
+        .. versionchanged:: 0.11.0
+            Log message format has changed from ``{"log": message}`` to
+            ``["log", message]``
+        """
+        if self.version < (0, 11, 0):
+            if message is None:
+                message = 'Error: attempting to log message of None'
+            if not isinstance(message, util.strbase):
+                message = json.encode(message)
+            res = {'log': message}
+        else:
+            if not isinstance(message, util.strbase):
+                message = json.encode(message)
+            res = ['log', message]
+        self.respond(res)
```

Comment:
I guess this will be better.
```diff
diff --git a/couchdb/server/__init__.py b/couchdb/server/__init__.py
index 1a64835..40a4328 100644
--- a/couchdb/server/__init__.py
+++ b/couchdb/server/__init__.py
@@ -241,15 +241,14 @@ class BaseQueryServer(object):
             Log message format has changed from ``{"log": message}`` to
             ``["log", message]``
         """
+        if message is None:
+            message = 'Error: attempting to log message of None'
+        if not isinstance(message, util.strbase):
+            message = json.encode(message)
+
         if self.version < (0, 11, 0):
-            if message is None:
-                message = 'Error: attempting to log message of None'
-            if not isinstance(message, util.strbase):
-                message = json.encode(message)
             res = {'log': message}
         else:
-            if not isinstance(message, util.strbase):
-                message = json.encode(message)
             res = ['log', message]
         self.respond(res)

diff --git a/couchdb/tests/server/qs.py b/couchdb/tests/server/qs.py
index 8513a2d..3354263 100644
--- a/couchdb/tests/server/qs.py
+++ b/couchdb/tests/server/qs.py
@@ -237,11 +237,11 @@ class BaseQueryServerTestCase(unittest.TestCase):

     def test_log_none_message(self):
         output = StringIO()
-        server = BaseQueryServer(version=(0, 9, 0), output=output)
+        server = BaseQueryServer(output=output)
         server.log(None)
         self.assertEqual(
             output.getvalue(),
-            u'{"log": "Error: attempting to log message of None"}\n'
+            u'["log", "Error: attempting to log message of None"]\n'
         )

     def test_log_newstyle(self):
```